test/common/http/http1/codec_impl_test.cc

```diff
@@ -700,6 +700,19 @@ TEST_F(Http1ClientConnectionImplTest, PrematureResponse) {
   EXPECT_THROW(codec_->dispatch(response), PrematureResponseException);
 }
 
+TEST_F(Http1ClientConnectionImplTest, EmptyBodyResponse) {
```

Comment:
We already have a test that covers 1xx, but can you add a test for maybe 200 with content_length zero also?